src/utils.c

```diff
@@ -204,10 +220,19 @@ bool set_add(set_t *set, uint32_t key)
  * @set: a pointer points to target set
  * @key: the key of the inserted entry
  */
+#if RV32_HAS(SYSTEM) && RV32_HAS(JIT)
+bool set_has(set_t *set, uint64_t key)
```

Comment:
You should explain why wider key length is required.

Reply:
It is at https://github.com/sysprog21/rv32emu/blob/b982d62a4c77ee44476f2f47de9aabbbf18ba27a/src/utils.h#L144-L150

Should I mention it again here?

Reply:
> Should I mention it again here?

Consider to utilize typedef and helper routines to unify.